libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -204,8 +207,13 @@ func (p *Provider) generateHints(event bus.Event) bus.Event {
 
 	if rawCont, ok := kubeMeta["container"]; ok {
 		container = rawCont.(common.MapStr)
-		e["docker"] = common.MapStr{
-			"container": container,
+		// This would end up adding a docker|rkt.container entry into the event. This would make sure
+		// that there is not an attempt to spin up a docker input for a rkt container and when a
+		// rkt input exists it would be natively supported.
+		if runtime, ok := container["runtime"]; ok {
+			e[runtime.(string)] = common.MapStr{
```

Comment:
I've second thoughts about our current field names. I think we should rename `docker.container.*` to `container.*` for 7.0. `container.runtime` will give you `docker`, `rkt` or any other.

For the time being, I'm ok with this I think, as removing the `docker` prefix would be a breaking change

Reply:
What do you think about this @vjsamuel? I'm thinking we could apply the change for autodiscover already, somehow duplicating `docker.container` == `container`.

Then in 7.0 we update metadata processor to use the new fields